src/qtcore/qt.js

```diff
@@ -13,13 +13,18 @@ global.Qt = {
   // Load file, parse and construct as Component (.qml)
   createComponent: function(name) {
     if (name in engine.components)
-        return engine.components[name];
+    {
+      var newComponentInstance = Object.create( engine.components[name] );
+      newComponentInstance.$context = _executionContext;
+      return newComponentInstance;
+    }
```

Comment:
This was revered in 871fbf03c8dcdda039c5556add2bdfb1b2523c41, see second comment there.